hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -1202,6 +1218,7 @@ public void closeContainer(Container container)
       }
       container.close();
       ContainerLogger.logClosed(container.getContainerData());
+      createContainerMerkleTree(container);
```

Comment:
This would cover only the clean close condition. I guess we need to discuss the various trigger for generating the merkle tree and the exact code path for it.

Reply:
Yes as per our discussion offline. I will add the merkle to generation to all the paths.